### DISCUSSION TITLE
Compatibility with 32-bit macOS

### DISCUSSION THREAD

**Discussion by gammaburst2024** [ReagentX/imessage-exporter#305] [Q&A] [answered]
Get the following error when trying to run for the first time:

```
dyld: cannot load 'imessage-exporter' (load command 0x80000034 is unknown)
Abort trap: 6
```

Running Mojave on Intel Core M

**Comment by ReagentX** [comment, marked as answer]
Your macOS version is too old to run the pre-compiled binary; you will need to compile `imessage-exporter` for your platform. It is worth noting that Rust dropped support for 32-bit macOS [in 2020](https://blog.rust-lang.org/2020/01/03/reducing-support-for-32-bit-apple-targets.html).

Alternatively, you may have downloaded the Apple Silicon build for an Intel Mac.
